marketdata/rest.go

```diff
@@ -1769,24 +1769,33 @@ func (c *Client) get(u *url.URL) (*http.Response, error) {
 }
 
 func unmarshal(resp *http.Response, v easyjson.Unmarshaler) error {
+	if resp == nil || resp.Body == nil {
+		return errors.New("response or response body is nil")
+	}
+
 	var (
-		reader io.ReadCloser
+		reader = resp.Body
 		err    error
 	)
 	switch resp.Header.Get("Content-Encoding") {
 	case "gzip":
 		reader, err = gzip.NewReader(resp.Body)
 		if err != nil {
+			resp.Body.Close() // Prevents resource leak if gzip.NewReader fails
```

Comment:
We should call `closeResp` every time in a defer before calling `unmarshal()`. Calling it here doesn't really make sense, because it won't cover the happy (non-nil) path. 

That being said, if you check all `unmarshal()` references, we actually don't call `closeResp()` before all of them. For example in `GetMultiTrades` I don't think it's called. Adding those missing calls instead would be great in this PR!